### PR TITLE
Fixes for retrying requests and token refreshes

### PR DESCRIPTION
**Pull request by miburf**
Fixes for retrying requests and token refreshes
Add tests for retrying

### CODE REVIEW

File: Samples/SDKFeatures/SDKFeatures/ViewControllers/ActionPlanTasks/MHVActionPlanTaskListViewController.m [hunk 1]
```diff
@@ -113,15 +113,14 @@ - (void)viewDidLoad
     
     self.tableView.dataSource = self;
     self.tableView.delegate = self;
-    
-    [self loadActionPlanTasks];
-    [self loadActionPlan];
```
Comment (mikenevi) [inline Samples/SDKFeatures/SDKFeatures/ViewControllers/ActionPlanTasks/MHVActionPlanTaskListViewController.m:118]: Thanks for fixing that :)

**Review by mikenevi — APPROVED** [review]

File: MHVLib/Connection/MHVConnection.m [hunk 2]
```diff
@@ -402,6 +445,20 @@ - (void)refreshSessionCredentialWithCompletion:(void(^_Nullable)(NSError *_Nulla
...
 {
+    // If it already refreshed the token for this request, don't retry since refreshing again won't fix the problem
+    if (request.hasRefreshedToken)
+    {
+        MHVLOG(@"The refreshed session authorization is invalid or expired.");
+        
+        if (request.completion)
+        {
+            request.completion(nil, [NSError error:[NSError MHVUnauthorizedError]
+                                   withDescription:@"The refreshed session authorization is invalid or expired."]);
```
Comment (namalu) [inline MHVLib/Connection/MHVConnection.m]: We should add to the description that an attempt to refresh the token was made but failed.

**Review by namalu — CHANGES_REQUESTED** [review]
I'm not seeing any updated tests to verify the new functionality added.

File: MHVLibTests/Connection/MHVConnectionTests.m [hunk 3]
```diff
@@ -412,16 +398,174 @@ @interface MHVConnection (Testing)
...
+                    
+                    MHVMethod *method = [MHVMethod getThings];
+                    method.parameters = @"GETTHINGSBODY";
+                    [testConnection executeHttpServiceOperation:method
+                                                     completion:^(MHVServiceResponse * _Nullable response, NSError * _Nullable error) { }];
+                });
+                
+                it(@"requestCount should be 3", ^
+                   {
+                       [[expectFutureValue(requestCount) shouldEventuallyBeforeTimingOutAfter(10)] equal:@(3)];
```
Comment (namalu) [inline MHVLibTests/Connection/MHVConnectionTests.m:426]: I think we also need to make sure that requests don't end up in an infinite loop. This test will succeed once the requestCount variable == 3, but if we ned to make sure it never reaches 4.

**Review by namalu — APPROVED** [review]

File: MHVLibTests/Connection/MHVConnectionTests.m (hunk 3 above)
Comment (miburf) [reply MHVLibTests/Connection/MHVConnectionTests.m:426]: Added a test that it doesn't reach 4

File: MHVLib/Connection/MHVConnection.m (hunk 2 above)
Comment (miburf) [reply MHVLib/Connection/MHVConnection.m]: Improved the description

**Pull request merged**